### DISCUSSION TITLE
project perspective and contributions

### DISCUSSION THREAD

**Discussion by dufoli** [tprouvot/Salesforce-Inspector-reloaded#246] [General]
Hello, I see that you have no time to review my PRs but you still work on other features.

May I ask why ?
- do you really want these features ?
- have you less time or less motivation to maintain ?
- do you need other maintainer to handle workload ?
- else I can do a fork with a kind of beta version with more features ?

I have maintain or be an active contributor few open source repo in the past in big project mostly( gnome, monotorrent, pinta, dolserver, quarkus, quarkus-cxf,...). So I know it take a lot of time. Especially review... 
but 2 releases happend and no feedback on my PR so that s why I ask here.
And it is not a critic but I try to be constructive to know if I continue to provide PR or start a new project with my stuff. I plan to refactor a lot the parser stuff which is totaly ugly (regex) to have a more standard way (tokenizer, lexer).

**Comment by tprouvot** [comment]
Hi @dufoli,

You're right I did not prioritize your PRs, my apologies about that.
I'm trying to implement / review new features with small changes and high business value (from my perspective, so it may be subjective) and avoid breaking changes.

Since few weeks and the release of the beta version I'm more comfortable with the fact that bugs may affect fewer users and changes are more largely tested.

Some of your PR have been created with draft status and because of that I did not considered it to be ready for review, after that some other PR came through and I my attention was focused on the new PRs.

Beside of that I'm communicating directly with some of the contributors (outside of git, with slack) and this facilitate the collaboration.
You can reach me on Linkedin if you want/can connect on slack.

I'm also thinking to add some colleagues as reviewer to help me on that.

I'll start to review https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/178 and https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/179 and hopefully find the time for https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/185.

Thanks for your contributions and apologies for the delay.

**Comment by dufoli** [comment]
About test, I totaly agree and that s why I want to integrate unit test in github action in order to have test run inside the pipeline of build like any other project. #201 
The current system for test is really shitty and I have not succeed to run it by following the process.
That is why I want to have unit test instead of integration test.
But it is a pain to do it with my current approach which is in DRAFT mode.
- unit test only controller js => impact is that code must be split between controller and view
- mock visual part (current approach), I am not sure about this solution because we have to mock a lot of object and there is an issue between import and require style.
- use a docker to run a browser to launch test

I have not seen any point about a slack.
I can contact you on linkedin if you want but I do not used it a lot.... I prefer to use slack too. Can you provide me the link ?

I used day to day the 178 + 179 because it is really a pain in the ass to move data from prod org to a sandbox in order to reproduce bug outside of production env because each time, you have to select entity and fix externalid link and remove object column.
I have a list of SOQL to get data with export but import take a lot of time...

About SOSL, I have manage few developper who only used SOQL because there is a lot of stuff to help but on SOSL quite no tool (except dev console) to help to build query and see result. So priority is lower but can help dev to avoid the bad practice to use SOQL with like on many fields...
